hack/make-rules/helpers/go2make/go2make.go

```
@@ -185,45 +184,41 @@ func (emit emitter) loadPackages(targets ...string) ([]*packages.Package, error)
 	return packages.Load(&cfg, targets...)
 }
 
-func (emit emitter) visitPackages(pkgs []*packages.Package) (map[string]*packages.Package, []packages.Error) {
+func (emit emitter) visitPackages(pkgs []*packages.Package) map[string]*packages.Package {
 	pkgMap := map[string]*packages.Package{}
+	errs := false
 	for _, p := range pkgs {
-		errs := emit.visitPackage(p, pkgMap)
-		if len(errs) > 0 {
-			return nil, errs
+		ok := emit.visitPackage(p, pkgMap)
+		if !ok {
+			errs = true
 		}
 	}
-	return pkgMap, nil
+	if errs {
+		return nil
+	}
+	return pkgMap
 }
 
-func (emit emitter) visitPackage(pkg *packages.Package, pkgMap map[string]*packages.Package) []packages.Error {
+func (emit emitter) visitPackage(pkg *packages.Package, pkgMap map[string]*packages.Package) bool {
 	debug("visiting package", pkg.PkgPath)
 	if pkgMap[pkg.PkgPath] == pkg {
 		debug("  ", pkg.PkgPath, "was already visited")
-		return nil
+		return true
 	}
+	debug("  ", pkg.PkgPath, "is new")
+	pkgMap[pkg.PkgPath] = pkg
 
-	if len(pkg.Errors) > 0 {
-		debug("  ", pkg.PkgPath, "has errors:")
-		errs := []packages.Error{}
-		for _, e := range pkg.Errors {
-			debug("    ", fmt.Sprintf("%q", e))
-			if e.Kind == packages.ListError {
-				// ignore errors like "build constraints exclude all Go files"
-				debug("    ignoring error")
-				continue
-			}
```

this error was ignored ALWAYS, don't we need to keep ignoring it?